src/python/pants/option/global_options.py

```diff
@@ -122,6 +122,12 @@ def register_bootstrap_options(cls, register):
     register('--pants-version', advanced=True, default=pants_version(),
              help='Use this pants version.')
 
+    # Like `--pants-version`, Pants code uses this only to verify that the user is using the requested interpreter
+    # version. It too may be grepped out of pants.ini to be used by things like setup scripts,
+    # runner scripts, IDE plugins, etc. in order to select which interpreter to use.
+    register('--pants-engine-python-version', advanced=True,
```

Comment:
IMO, just `pants_python_version` would be sufficient here, because global options are namespaced, and are not recursive into other subsystem namespaces.

But I would move more of what is in the comment into the help for this option (and if you're feeling ambitious, maybe do the same help expansion for `--pants-version`)?

Reply:
Moved the comment entirely into help, along with for `--pants-version`! Thanks for the suggestion.

Regarding `pants_python_version`, I'm concerned that it wouldn't be clear enough this is what Pants uses for itself and is not what Pants uses for running the user's own code. I want to make sure it's crystal clear this is different than `--python-setup-interpreter-constraints`. Also not super concerned about the option taking extra characters to type, as this is supposed to be pinned in `pants.ini`. But definitely fine with renaming if others prefer `pants_python_version`! Thoughts from others?